File: server/shared/types/order.go
package types

type ItemsType []struct {
	MenuID   int32   `json:"menu_item_id"`
	Quantity int32   `json:"quantity"`
	Price    float32 `json:"price"`
	Category string  `json:"category"`
}

type OrderPayload struct {
	TableNumber string    `json:"table_number"`
	Items       ItemsType `json:"items" binding:"required"`
	Total       float32   `json:"total_price" binding:"required"`
	Note        *string   `json:"note"`
}

type OrderEditPayload struct {
	TableNumber *string    `json:"table_number"`
	Items       *ItemsType `json:"items"`
	Total       *float32   `json:"total_price"`
	Note        *string    `json:"note"`
}

File: web/src/types/order.ts
export interface Order {
  id: number
  table_number: string
  status: string
  total_price: number
  note?: string
  waiter_id?: number
  waiter_name?: string
  created_at: string
  updated_at: string
  delivered_at?: string
}

export interface ItemType {
  menu_item_id: number
  quantity: number
}

export interface OrderPayload {
  table_number: string
  total_price: number
  items: Array<ItemType>
  note?: string
}

export interface OrderItem {
  id: number
  order_id: number
  menu_item_id: number
  name: string
  quantity: number
  price: number
}
